include/exadg/incompressible_navier_stokes/spatial_discretization/operator_dual_splitting.cpp

```diff
@@ -596,64 +596,126 @@ OperatorDualSplitting<dim, Number>::local_interpolate_velocity_dirichlet_bc_boun
   Range const & face_range) const
 {
   unsigned int const dof_index = this->get_dof_index_velocity();
-  AssertThrow(
-    matrix_free.get_dof_handler(dof_index).get_triangulation().all_reference_cells_are_hyper_cube(),
-    dealii::ExcMessage("This function is only implemented for hypercube elements."));
-  unsigned int const quad_index = this->get_quad_index_velocity_gauss_lobatto();
 
-  FaceIntegratorU integrator(matrix_free, true, dof_index, quad_index);
-
-  for(unsigned int face = face_range.first; face < face_range.second; face++)
+  if(matrix_free.get_dof_handler(dof_index)
+       .get_triangulation()
+       .all_reference_cells_are_hyper_cube())
   {
-    dealii::types::boundary_id const boundary_id = matrix_free.get_boundary_id(face);
+    unsigned int const quad_index = this->get_quad_index_velocity_gauss_lobatto();
```

Comment:
By introducing a member function `get_quad-index_velocity_nodal_points()` we can unify the hypercube/simplex case and write this line outside the if-else-statement for hypercube vs. simplex.

Reply:
Good idea, I like the name.